### PR TITLE
Added config file

### PR DESCRIPTION
**Pull request by Tangdongle**
Added new option --createRcFile to generate/reset the config file
History can now be set to per-session or shared (default)

### CODE REVIEW

**Comment by 0atman** [comment]
Incredible! Could you update the readme as well? Maybe add a config section with keys, location of the config file?

**Comment by Tangdongle** [comment]
No worries, readme updated

File: README.md [hunk 1]
```diff
@@ -14,5 +14,16 @@
...
 * Colorized output
 
+## Config
+Config is saved and loaded from `configDir / inim`.
+* On Windows, this is %APPDATA%\inim
+* On Linux, this is /home/<user>/.config/inim
+
+Currently, the config allows you to set two options:
+* Style
+  * Set prompt string (default: true)
```
Comment (0atman) [inline README.md:24]: default "inim>" I think?

**Comment by 0atman** [comment]
Config looks good - I've tested both prompt and history options and they seem to work :ok_hand: 

Ie, when inimrc is

```ini
[History]
persistent=True
[Style]
prompt="nimmer> "
```

The prompt changes:

```
./inim                                                              
👑 INim 0.4.5                                                                               
Nim Compiler Version 1.2.0 [Linux: amd64] at /home/oatman/.nimble/bin/nim                   
nimmer>                                                                                     
                                                                                            
Quitting INim: Goodbye!  
```

File: src/inim.nim [hunk 2]
```diff
@@ -50,10 +65,8 @@ var
 
 when promptHistory:
     # When prompt history is enabled, we want to load history
-    # TODO: Config to allow per-session history
-    var historyFile =  getConfigDir() / "inim"
-    discard existsorCreateDir(historyFile)
-    historyFile = historyFile / ".history"
+    var historyFile = if config.getSectionValue("History", "persistent") == "True": ConfigDir / "history"
```
Comment (0atman) [inline src/inim.nim:68]: What do you think about calling the file `history.nim`? It mostly contains Nim code, right? Might make it nicer to syntax highlight when reading.

File: src/inim.nim [hunk 3]
```diff
@@ -23,10 +25,23 @@ const
     ]
     # preloaded code into user's session
     EmbeddedCode = staticRead("inimpkg/embedded.nim")
+    ConfigDir = getConfigDir() / "inim"
+    RcFilePath = ConfigDir / "inimrc"
```
Comment (0atman) [inline src/inim.nim:29]: Should we call it `inim.ini`, as _"The configuration file's syntax is similar to the Windows .ini format"_, and then we'll get syntax highlighting for free?

File: src/inim.nim [hunk 4]
```diff
@@ -23,10 +25,23 @@ const
     ]
     # preloaded code into user's session
     EmbeddedCode = staticRead("inimpkg/embedded.nim")
+    ConfigDir = getConfigDir() / "inim"
+    RcFilePath = ConfigDir / "inim.ini"
```
Comment (0atman) [inline src/inim.nim:29]: Nice!

File: src/inim.nim (hunk 2 above)
Comment (Tangdongle) [reply src/inim.nim]: Good call. If we want external editor support down the track, it's likely going to be reading that file

File: src/inim.nim [hunk 5]
```diff
@@ -50,10 +65,8 @@ var
 
 when promptHistory:
     # When prompt history is enabled, we want to load history
-    # TODO: Config to allow per-session history
-    var historyFile =  getConfigDir() / "inim"
-    discard existsorCreateDir(historyFile)
-    historyFile = historyFile / ".history"
+    var historyFile = if config.getSectionValue("History", "persistent") == "True": ConfigDir / "history.nim"
```
Comment (0atman) [inline src/inim.nim:68]: very nice

File: src/inim.nim (hunk 2 above)
Comment (0atman) [reply src/inim.nim]: :ok_hand:

**Comment by Tangdongle** [comment]
Yeah, all good to merge. Should be able to start adding more features now, especially opt-in/out features

**Comment by 0atman** [comment]
sweet, many thanks!

**Comment by 0atman** [comment]
fixes #61

**Pull request merged**